test/fixtures/schema.sql

```diff
@@ -175,6 +175,14 @@ create view orders_view as
 create view consumers_view as
   select * from public.public_consumers;
 
+create view consumers_view_view as
+  select * from consumers_view;
+
+create view public.consumers_extra as
+  select * from consumers_view;
+
+create view consumers_extra_view as
+  select * from public.consumers_extra;
```

Comment:
@wolfgangwalther Hmm.. still a bit confused about this test setup, perhaps you can clarify. This is what I'm seeing:

- `public.consumers_extra` - the middle(or hidden) view.
- `test.consumers_extra_view` - the exposed view which is based on the middle view.

So far so good. The problem is that the middle view `public.consumers_extra` is based on `test.consumers_view`(filling the explicit schema here) which is is already on a exposed schema.

That doesn't look like the inside-out approach we discussed before. It's more like an in-out-in-out, if that makes sense.

This is a test setup that makes sense to me:

```sql
create view public.consumers_middle as
select * from public.public_consumers;
                                          
create view test.consumers_middle_view as
select * from public.consumers_middle;
```

And the test:

```http
GET /consumers_hidden_view?select=*,orders_view(*)
```

I've tested it and it works good. What do you think? Is the test clearer that way?

Reply:
I've also checked that the above test setup didn't work on your first recursive-only approach. That commit got lost because of the force-push, but the git reflog helped me there :). So the middle view query is definitely good, only want to clarify the test before merging :rocket: .

Reply:
> I've tested it and it works good. What do you think? Is the test clearer that way?

We currently have the following two view chains exposed on the public schema [view name: chain of schema types]:

- consumers_view_view: exposed -> exposed > hidden
- consumers_extra_view: exposed -> extra -> exposed -> hidden

As you noted only the "exposed -> extra" part is relevant for this test-case.

The only reason why I based the extra view on another public view was, because this allows us to implicitly test another level of recursion. So the 2nd chain is now longer compared to the first one. I didn't feel like adding another test-case just for this, so I took the short way out here.

Reply:
> That commit got lost because of the force-push, but the git reflog helped me there :).

I might have to force-push less... but first I have to see what that reflog is ;)

Reply:
> The only reason why I based the extra view on another public view was, because this allows us to implicitly test another level of recursion

Ahh, I see. It's more of an artificial test. Then it's all good! I'll merge :rocket:

Reply:
> I might have to force-push less... but first I have to see what that reflog is ;)

I'm guilty of force-pushing as well :grin: . So no need to change any workflows, reflog is good enough :)